src/datetime/tests.rs

```diff
@@ -857,6 +857,15 @@ fn test_datetime_from_local() {
     assert_eq!(datetime_west, datetime_utc.with_timezone(&timezone_west));
 }
 
+#[test]
+fn test_issue_651() {
+    // dt corresponds to `FILETIME = 147221225472` from the issue.
+    // This used to fail on Windows for timezones with an offset of -5:00 or greater.
+    let dt = NaiveDate::from_ymd_opt(1601, 1, 1).unwrap().and_hms_milli_opt(4, 5, 22, 122).unwrap();
+    let local_dt = Local.from_utc_datetime(&dt);
+    dbg!(local_dt);
+}
+
```

Comment:
Would it be of any added value to check the leap year information for 1500 and 1600 on windows?

Reply:
Handling milliseconds and dealing with extreme years where you have to account for the 400-year leap cycle were your efforts in https://github.com/chronotope/chrono/pull/997. I am sorry this PR crosses that and takes a different approach.

We are not doing anything special anymore in that area, it should be as valid as other leap year handling in chrono.